numba/targets/cpu.py

```diff
@@ -186,6 +186,41 @@ def calc_array_sizeof(self, ndim):
         return self.get_abi_sizeof(self.get_value_type(aryty))
 
 
+class FastMathOptions(object):
+    """
+    Options for controlling fast math optimization.
+    """
+    def __init__(self, value):
```

Comment:
The exception handling over the various input types needs testing in the unit tests for this feature. This sort of pattern would likely be appropriate (i.e. check exception type and message):

https://github.com/numba/numba/blob/0ede0b7e8dc3ec8f7d22210573fbf268bf0d68f6/numba/tests/test_dictimpl.py#L533-L538